src/main/scala/io/circe/yaml/parser/package.scala

```diff
@@ -76,26 +76,29 @@ package object parser {
       case _ => Left(ParsingFailure("Only string keys can be represented in JSON", null))
     }
 
-    node match {
-      case mapping: MappingNode =>
-        flattener.flatten(mapping).getValue.asScala.foldLeft(
-          Either.right[ParsingFailure, JsonObject](JsonObject.empty)
-        ) {
-          (objEither, tup) => for {
-            obj <- objEither
-            key <- convertKeyNode(tup.getKeyNode)
-            value <- yamlToJson(tup.getValueNode)
-          } yield obj.add(key, value)
-        }.map(Json.fromJsonObject)
-      case sequence: SequenceNode =>
-        sequence.getValue.asScala.foldLeft(Either.right[ParsingFailure, List[Json]](List.empty[Json])) {
-          (arrEither, node) => for {
-            arr <- arrEither
-            value <- yamlToJson(node)
-          } yield value :: arr
-        }.map(arr => Json.fromValues(arr.reverse))
-      case scalar: ScalarNode => convertScalarNode(scalar)
+    if (node == null) {
```

Comment:
Sorry to nitpick, but are there other cases where SnakeYAML will return `null` here?

Reply:
As for my knowledge - for blank or empty strings, for `null` string it returns a `org.yaml.snakeyaml.nodes.ScalarNode`

The code below passes.

```scala
implicit def str2Reader(s: String): Reader = new StringReader(s)

val parser = new Yaml()
parser.compose("") shouldBe null
parser.compose("  ") shouldBe null
parser.compose("\n") shouldBe null
parser.compose("\r\n") shouldBe null
```

Passing null to the parser results in NullPointerException.